### PR TITLE
Add tests

### PR DESCRIPTION
**Pull request by luodan01**
Add tests to increase code coverage

### CODE REVIEW

File: src/main/java/seedu/anilist/model/ModelManager.java [hunk 1]
```diff
@@ -42,6 +42,7 @@ public ModelManager(ReadOnlyAnimeList animeList, ReadOnlyUserPrefs userPrefs) {
         this.filteredAnime = new FilteredList<>(this.animeList.getAnimeList());
         this.currentTab = new TabOption("all");
         this.tabOptionFilter = PREDICATE_SHOW_ALL_ANIME;
+        updateUserStats();
```
Comment (muhammad-faruq) [inline src/main/java/seedu/anilist/model/ModelManager.java:45]: Why is this suddenly needed?

File: src/main/java/seedu/anilist/ui/TabOption.java [hunk 2]
```diff
@@ -69,7 +67,7 @@ private TabOptions parseTabStatus(String tabString) {
     @Override
     public boolean equals(Object other) {
         return other == this // short circuit if same object
-                || (other instanceof Status // instanceof handles nulls
+                || (other instanceof TabOption // instanceof handles nulls
```
Comment (muhammad-faruq) [inline src/main/java/seedu/anilist/ui/TabOption.java:70]: Shouldn't this follow the same as the previous test where you assert != null?

File: src/test/java/seedu/anilist/model/stats/StatsTest.java [hunk 3]
```diff
@@ -112,4 +118,28 @@ public void getStats_withAllGenresInGenreList_returnsTrue() {
...
+                put(firstGenre, 2);
+            }};
+        Stats stats1 = new Stats(1, 2, 3, genres1);
+        Stats stats2 = new Stats(1, 2, 3, genres2);
+        Stats stats3 = new Stats(0, 0, 0, new HashMap<>());
+
+        //same Stats object
+        assertTrue(stats1.equals(stats1));
+        //different Stats object
+        assertTrue(stats1.equals(stats2));
```
Comment (muhammad-faruq) [inline src/test/java/seedu/anilist/model/stats/StatsTest.java:141]: can the description be better?

**Review by muhammad-faruq — COMMENTED** [review]
Some minor clarifications.. other than that LGTM

File: src/main/java/seedu/anilist/model/ModelManager.java (hunk 1 above)
Comment (luodan01) [reply src/main/java/seedu/anilist/model/ModelManager.java:45]: I realised while writing tests that Stats is the only field that's uninitialised, so I just initialised it here. makes testing easier too

File: src/test/java/seedu/anilist/model/stats/StatsTest.java (hunk 3 above)
Comment (luodan01) [reply src/test/java/seedu/anilist/model/stats/StatsTest.java:141]: Are you referring to something like //differe

File: src/main/java/seedu/anilist/ui/TabOption.java (hunk 2 above)
Comment (luodan01) [reply src/main/java/seedu/anilist/ui/TabOption.java:70]: If you mean the one in `Stats.java` I will be removing that

File: src/test/java/seedu/anilist/model/stats/StatsTest.java (hunk 3 above)
Comment (muhammad-faruq) [reply src/test/java/seedu/anilist/model/stats/StatsTest.java:141]: yeap. so like `//different stats object but identical genre count and episode count`

**Review by muhammad-faruq — APPROVED** [review]

**Pull request merged**